### PR TITLE
[Opsgenie] - Add Action to `Unacknowledge Alert` + modernize + EmptyViews + Optimized Updates + Caching

### PR DESCRIPTION
**Pull request by xmok**
## Description

CHANGELOG
### Enhancements
- Alerts can now be Unacknowledged
- EmptyViews are shown based on your query
- Toasts are shown before, during and after actions
- Opsgenie batches some requests so e.g. after closing an alert, the alert may still be shown as "open" until you reload/retype the Raycast query; now the updated result is shown locally so you can reasonably know what the current status is without having to refresh (though fetching lastest data is always recommended)

### Dev
- Remove Preferences type
- Imrpove Error handling and data fetching by declaring their types
- utilize `useFetch for GET requests
- utilize **mutate** of `useFetch` for POST requests

## Screencast

The screencast shows a temporary Opsgenie instance created for this screencast.

https://github.com/user-attachments/assets/3f65f8de-775e-4038-acb8-559d1edac555

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/opsgenie/README.md [hunk 1]
```diff
@@ -1,3 +1,30 @@
 # Opsgenie
 
-The Opsgenie extension can be used to access alerts and incidents from [Opsgenie](https://www.atlassian.com/software/opsgenie). The extension can also used to acknowledge, snooze and close alerts.
+The Opsgenie extension can be used to access alerts and incidents from [Opsgenie](https://www.atlassian.com/software/opsgenie). The extension can also used to acknowledge, unacknowledge, snooze and close alerts.
+
+## 🚀 Getting Started
+You will need the following:
+
+1. API URL of Opsgenie instance - this will depend on where your instance is based-
```
Comment (ricoberger) [inline extensions/opsgenie/README.md:8]: Should the two available urls be added here?

- https://api.opsgenie.com
- https://api.eu.opsgenie.com/

File: extensions/opsgenie/README.md [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+2. API Key - for best results, you will need an "Integration" API Key:
+    - `Navigate` to your Opsgenie instance
+    - `Navigate` to **Teams**
+    - `Open` the Team
+    - `Open` **Integrations**
+    - `Click` **Add integration**
+    - `Select` "API"
+    - `Generate` the key
+    - (The options may vary so refer to https://support.atlassian.com/opsgenie/docs/create-a-default-api-integration/)
+3. Opsgenie Usernam - this is the username you want to use for alerts/incidents
```
Comment (ricoberger) [inline extensions/opsgenie/README.md:18]:
```suggestion
3. Opsgenie Username - this is the username you want to use for alerts/incidents
```

**Review by ricoberger — APPROVED** [review]
lgtm, awesome work 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**